In [62]:
from sklearn.datasets import load_iris
iris = load_iris()
dir(iris)

['DESCR',
 'data',
 'data_module',
 'feature_names',
 'filename',
 'frame',
 'target',
 'target_names']

In [63]:
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import cross_val_score

In [64]:
import numpy as np

In [65]:
np.average(cross_val_score(LogisticRegression(max_iter=200), iris.data, iris.target))

0.9733333333333334

In [66]:
np.average(cross_val_score(LogisticRegression(max_iter=200, C=100), iris.data, iris.target))

0.9733333333333334

In [67]:
np.average(cross_val_score(DecisionTreeClassifier(), iris.data, iris.target))

0.9533333333333334

In [68]:
np.average(cross_val_score(DecisionTreeClassifier(criterion='entropy'), iris.data, iris.target))

0.9533333333333334

In [69]:
np.average(cross_val_score(SVC(), iris.data, iris.target))

0.9666666666666666

In [70]:
np.average(cross_val_score(SVC(kernel='linear'), iris.data, iris.target))

0.9800000000000001

In [71]:
np.average(cross_val_score(SVC(kernel='linear', C=0.6), iris.data, iris.target))

0.9866666666666667

In [72]:
np.average(cross_val_score(RandomForestClassifier(), iris.data, iris.target))

0.96

In [73]:
np.average(cross_val_score(RandomForestClassifier(n_estimators=50), iris.data, iris.target))

0.96

Best one is **SVC(kernel='linear', C=0.6)** with the score of **0.9867**.